### PR TITLE
Adds a simple class that watches an executor for being blocked.

### PR DESCRIPTION
**Pull request by balazsracz**

### CODE REVIEW

File: src/utils/ExecutorWatchdog.hxx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    ExecutorWatchdog(Service *service)
+        : StateFlowBase(service)
+    {
+        start_flow(STATE(take_stamp));
+    }
+
+private:
+    Action take_stamp()
+    {
+        lastTimeMsec_ = os_get_time_monotonic() / 1000000;
```
Comment (atanisoft) [inline src/utils/ExecutorWatchdog.hxx:59]: should we use `NSEC_TO_MSEC`?

File: src/utils/ExecutorWatchdog.hxx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        return sleep_and_call(&timer_, MSEC_TO_NSEC(50), STATE(woken));
+    }
+
+    Action woken()
+    {
+        uint32_t new_time_msec = os_get_time_monotonic() / 1000000;
```
Comment (atanisoft) [inline src/utils/ExecutorWatchdog.hxx:65]: similar here for `NSEC_TO_MSEC`

File: src/utils/ExecutorWatchdog.hxx (hunk 1 above)
Comment (balazsracz) [reply src/utils/ExecutorWatchdog.hxx]: done

File: src/utils/ExecutorWatchdog.hxx (hunk 2 above)
Comment (balazsracz) [reply src/utils/ExecutorWatchdog.hxx]: done

**Pull request merged**